arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java

```diff
@@ -168,37 +169,44 @@ public SelfdestructSection(Hub hub, MessageFrame frame) {
     // - The recipient address will become warm (i+3)
     //   * recipientFirstAccountFragment
 
-    selfdestructorAccountAfter = selfdestructorAccountBefore.deepCopy().setBalanceToZero();
+    selfdestructorNew = selfdestructor.deepCopy().setBalanceToZero();
+
+    final boolean isDeployment = frame.getType() == MessageFrame.Type.CONTRACT_CREATION;
+    checkState(isDeployment == selfdestructor.deploymentStatus());
+    if (isDeployment) {
+      selfdestructorNew = selfdestructorNew.deploymentStatus(false);
+      selfdestructorNew.code(Bytecode.EMPTY);
+    }
```

Comment:
Unexceptional `SELFDESTRUCT`'s in deployment contexts trigger an empty deployment. Thus we reset the byte code (from init code to `Bytes.EMPTY`.)